res_users_password_security/README.rst

```diff
@@ -0,0 +1,94 @@
+.. image:: https://img.shields.io/badge/license-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+==================
+Password  Security
+==================
+
+This module allows admin to set company-level password security requirements
+and enforces them on the user.
+
+It contains features such as
+
+* Password expiration days
+* Password length requirement
+* Password minimum number of lowercase letters
+* Password minimum number of uppercase letters
+* Password minimum number of numbers
+* Password minimum number of special characters
```

Comment:
modern hashing algorithms don't only depend on the input string, but also a salt (another input string) and the amount of rounds: https://pythonhosted.org/passlib/lib/passlib.hash.pbkdf2_digest.html#interface
Which is a good thing, and you shouldn't make an attempt to fix the salt or whatever. Just take the strings as they are, and try new passwords against them. The result contains those values, so you don't need to worry about it: https://pythonhosted.org/passlib/lib/passlib.hash.pbkdf2_digest.html#format-algorithm